### ISSUE TITLE
Nan value

### ISSUE THREAD

**Issue by weichen456** [tensorflow/tpu#795]
Hello, I have retrain the SpineNet143 - retinanet model in TPUv3-32 devices batch size=128, lr=0.14, but I  encounted this issues: Tensor had NaN values in the 30000 train step. Meanwhile,  I have decreased the lr, but the result is not well. Would you please tell me what devices you used to train this model ? 
Thank you!

**Comment by xianzhidu** [comment]
We used 4x8 TPU devices with batch size 256 and lr 0.28 to train. Did you turn on the synchronized batch norm during model training? Another option is to train SpineNet with MaskRCNN, it is very unlikely to hit NaN.

**Comment by weichen456** [comment]
> We used 4x8 TPU devices with batch size 256 and lr 0.28 to train. Did you turn on the synchronized batch norm during model training? Another option is to train SpineNet with MaskRCNN, it is very unlikely to hit NaN.

Of course,  I haveturned on the synchronized batch norm during model training.  TPU4X8 devices?, why I can't find the detail information about it?  Dose it your internal device type?
Ok, I will try the MaskRCNN.
